pkg/controllers/queue/queue_controller_action.go

```diff
@@ -42,9 +61,11 @@ func (c *queuecontroller) syncQueue(queue *schedulingv1beta1.Queue, updateStateF
 		// Ignore error here, tt can not occur.
 		ns, name, _ := cache.SplitMetaNamespaceKey(pgKey)
 
-		// TODO: check NotFound error and sync local cache.
 		pg, err := c.pgLister.PodGroups(ns).Get(name)
 		if err != nil {
+			if errors.IsNotFound(err) {
```

Comment:
The err will also be returned and re-enqueued, and next procesee will continue handle the req, how about returned nil when pg not existed?
```go
		pg, err := c.pgLister.PodGroups(ns).Get(name)
		if err != nil {
			if !errors.IsNotFound(err) {
				return err
			}
			c.pgMutex.Lock()
			defer c.pgMutex.Unlock()

			delete(c.podGroups[queue], pgKey)
			return nil
		}
```

Reply:
Hi @Monokaix Thanks for your reviewing and sample code. In this case, I think we should 1) pop this error, so the error will be recorded and noticed by developers, and 2) re-enqueue the request, since the `syncQueue` process is actually not finished

Reply:
That's a good point, but what about just `continue` if pg not exists? And the queue will still be processed, re-enqueue seems is not so necessary.

Reply:
If we update cache silently and continue the process, the error will not be recorded in `handleQueueErr`, so the developers will not notice that in the events of queue. Sometimes we may need to monitor these events and do some alerting work

Reply:
Here we should make clear whether pg not exists is a normal case or else, if pg not exists, it's a normal case and we don't need to record an event and just print the err is ok, if you think it's an abnormal case, can you support more detail about why the pg is deleted when syncQueue?

Reply:
@Monokaix I think it is abnormal, because we have already added the event listener for pg events, including `Delete`:  
```
pgInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
		AddFunc:    c.addPodGroup,
		UpdateFunc: c.updatePodGroup,
		DeleteFunc: c.deletePodGroup,
	})
```
But it still missed this event and the `podGroups` cache is not updated. May be due to network issue. I think the case is rare and not easy to reproduce, but the possibility exists, as can be seen from my attached picture

Reply:
We can assume that `deletePodGroup` is a normal behavior，and what abnormal is that we didn't get the latest delete event through informer，based on this，we can skip process the pg because it truly not exists, and this will not block queue process and the queue will be synced.

Reply:
Whether get the event is not so important because the pg is not existed, and we just guarantee the queue is synced   is ok.

Reply:
@Monokaix Thanks for your advice, I update code to not pop the error and sync cache immediately, please have a look